zebrad/src/components/mempool/downloads.rs

```diff
@@ -238,19 +268,25 @@ where
             }?;
             let height = (height + 1).ok_or_else(|| eyre!("no next height"))?;
 
-            let tx = if let zn::Response::Transactions(txs) = network
-                .oneshot(zn::Request::TransactionsById(
-                    std::iter::once(txid).collect(),
-                ))
-                .await?
-            {
-                txs.into_iter()
-                    .next()
-                    .expect("successful response has the transaction in it")
-            } else {
-                unreachable!("wrong response to transaction request");
+            let tx = match gossiped_tx {
+                GossipedTx::Id(txid) => {
+                    let req = zn::Request::TransactionsById(std::iter::once(txid).collect());
+
+                    let tx = match network.oneshot(req).await? {
+                        zn::Response::Transactions(mut txs) => txs
+                            .pop()
+                            .expect("successful response has the transaction in it"),
+                        _ => unreachable!("wrong response to transaction request"),
+                    };
+
```

Comment:
@conradoplg I tidied this up a bit, what do you think?

Reply:
Oops, sorry, i didn't knew you were still working on it. I removed the auto merge.

Reply:
I like it, much better, thanks!